File: FactorialOfNumber/Fact.py
"""
    Author : Pakshal Jain
    Date : 23/10/2022
    Program : To Calculate Factorial of Entered Number
"""

#Function To Calculate Factorial
def Factorial(No) :
    i = 0;
    fact = 1;

    #Handeling Condition if Entered Number is Negative
    if(No < 0) :
        No = -No;

    #Logic
    for i in range(No,0,-1) :
        fact = fact * i;
    
    return fact;
def main() :
    No = 0;
    iret = 0;

    print("Jay Ganesh...");

    print("Enter One Number...");
    No = int(input());
    
    #Call To Factorial Function
    iret = Factorial(No);

    print("Factorial of Entered Number is :",iret);

#Call Back Method To Call Main Function
if __name__ =="__main__" :
    main();

File: FactorialOfNumberRecursiveApproach/Fact.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 08/05/2024
    Program : Write a recursive program which accept number from user and return its
              factorial.
              
              Input : 5
              Output : 120 
*/

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Required Libraries
package main;
import "fmt";

var imult int = 1;

////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Calculate Factorial of Number
func Factorial(ino int) int {
	if(ino != 0) {
		imult = imult * ino;
		ino--;
		Factorial(ino);
	} 
	
	return imult;
}

///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main Function
func main() {
	var ino int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh....");

	//Taking Input From User
	fmt.Println("Enter One Number : ");
	fmt.Scan(&ino);

	//Call To Factorial Function
	iret = Factorial(ino);
	
	//Printing Factorial of Number Calculated
	fmt.Printf("Factorial of %d  is : %d",ino,iret);
}